### PR TITLE
Fix bug in suiteapp file cabinet where folder parent id is unknown

### PR DESCRIPTION
**Pull request by dantal4**
---

_Additional context for reviewer_

---
_Release Notes_: 
Netsuite Adapter:
- Fix bug in suiteapp file cabinet where folder parent id is unknown
---
_User Notifications_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/suiteapp_file_cabinet.ts [hunk 1]
```diff
@@ -281,21 +298,11 @@ SuiteAppFileCabinetOperations => {
...
-    return [...getFullPath(idToFolder[folder.parent], idToFolder), folder.name]
-  }
-
-  const isMissingFolderParent = (folder: FolderResult, idToFolder: Record<string, FolderResult>): boolean => {
-    if (folder.parent !== undefined && idToFolder[folder.parent] === undefined) {
-      log.warn('folder parent does not exist: %o', folder)
-      return true
+    if (idToFolder[folder.parent] === undefined) {
+      log.error('folder\'s parent is unknown\nfolder: %o\nidToFolder: %o', folder, idToFolder)
+      throw new Error(`Failed to get absolute folder path of ${folder.name}`)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/suiteapp_file_cabinet.ts:303]: These is generally shouldn't happen, right? you already suppose to filter this case out in `queryFileCabinet`.

File: packages/netsuite-adapter/src/suiteapp_file_cabinet.ts [hunk 2]
```diff
@@ -271,7 +271,24 @@ SuiteAppFileCabinetOperations => {
...
+          }
+          return true
+        }),
+        filesResults: filesResults.filter(file => {
+          if (!folderIdsSet.has(file.folder)) {
+            log.warn('file\'s folder does not exist: %o', file)
+            return false
+          }
+          return true
+        }),
```
Comment (omrilit) [inline packages/netsuite-adapter/src/suiteapp_file_cabinet.ts:290]: I think that this code will not filter out all of the relevant cases.
Think of the situation of having the following files structure:`A/B/C/script.js`.
Assuming the `queryFiles` result is:
```
foldersResults: ['B', 'C']
filesResults: ['script.js']
``` 

In this case you will only filter out B but not all of its descendants.
Am I missing something?

File: packages/netsuite-adapter/src/suiteapp_file_cabinet.ts (hunk 1 above)
Comment (dantal4) [reply packages/netsuite-adapter/src/suiteapp_file_cabinet.ts:303]: Correct

File: packages/netsuite-adapter/src/suiteapp_file_cabinet.ts (hunk 2 above)
Comment (dantal4) [reply packages/netsuite-adapter/src/suiteapp_file_cabinet.ts:290]: You're rigth. That's probably the reason that my fix didn't work the last time.
I'll handle this case.

**Review by omrilit — APPROVED** [review]

**Pull request merged**